packages/core/contracts/zodiac/OptimisticGovernor.sol

```diff
@@ -231,19 +260,26 @@ contract OptimisticGovernor is Module, Lockable {
         }
     }
 
+    /**
+     * @notice Method to allow the owner to delete a particular proposal.
+     * @param _proposalId the id of the proposal being deleted.
+     */
     function deleteProposal(uint256 _proposalId) public onlyOwner {
-        // Delete a proposal that governance decided not to execute.
         delete proposalHashes[_proposalId];
         emit ProposalDeleted(_proposalId);
     }
 
-    function deleteRejectedProposal(
-        uint256 _proposalId,
-        uint256 _originalTime,
-        bytes memory _ancillaryData
-    ) public {
+    /**
+     * @notice Method to allow anyone to delete a proposal that was rejected.
+     * @param _proposalId the id of the proposal being deleted.
+     * @param _originalTime the id of the proposal being deleted.
+     */
+    function deleteRejectedProposal(uint256 _proposalId, uint256 _originalTime) public {
+        // Construct the ancillary data.
+        bytes memory ancillaryData = AncillaryData.appendKeyValueUint("", "id", _proposalId);
```

Comment:
Without this, there's no guarantee that the ancillary data matches the proposal id, and any rejected proposal would allow anyone to delete any proposal.